src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -292,7 +293,35 @@ private void OnRename(HubServiceEndpoint endpoint)
             {
                 return;
             }
-            // TODO: update local store names
+
+            Log.StartRenamingEndpoint(_logger, endpoint.Endpoint);
+            var oldEndpoint = _connectionContainers.FirstOrDefault(e => e.Key.Endpoint == endpoint.Endpoint).Key;
+
+            if (oldEndpoint != null)
+            {
+                _connectionContainers.TryRemove(oldEndpoint, out var container);

```

Comment:
> _connectionContainers.TryRemove(oldEndpoint, out var container); [](start = 15, length = 65)

After this is executed, messages routed to this `oldEndpoint` is lost. I think it is ok to keep this entry and don't remove it?

Reply:
If not remove, there'll be duplicated endpoints in cache and leads to duplicated messages like endpoint name not specified when broadcasts. Since container is no change, so old message should be able to processed and new messages should be able to route by new names. 

---
In reply to: [386811805](https://github.com/Azure/azure-signalr/pull/836#discussion_r386811805) [](ancestors = 386811805)